docs/select-syntax.md

```diff
@@ -0,0 +1,38 @@
+The `select` operation chooses between two values based on a simple string 
```

Comment:
Since we support quite a lot of different types, I think we shouldn't just expect a string.

We should support:
- If an array, empty array is false, otherwise true.
- If null, it is false
- If an empty string, it should be false
- If boolean, a boolean
- Of a function/operation/subroutine, the result of calling it
- If an object, true
- If a promise, we should wait for it and use the result
- ...

We would also need `equals` and other new primitives